### PR TITLE
PT-321: Better slack server

### PR DESCRIPTION
**Pull request by florianmski**
> Tracked in JIRA by [PT-321](https://novoda.atlassian.net/browse/PT-321)

### Scope of the PR

Slacker now keeps all the slack message that happened this day and clear them the next day. This way we have a more precise "biggest slacker" and don't miss "thanks" and "gallery" messages because of high turnover (before it was only keeping the last 100 messages in memory).

Also add a script to start automatically server and client (not sure how this would work on OSX, sorry).

### CODE REVIEW

File: node-server/server/widget/slacker/slacker.js [hunk 1]
```diff
@@ -28,14 +27,22 @@ function messageHandler(self, callback) {
...
+    self.messages.push(message);
   }
 }
 
+function isSameDayFor(message1, message2) {
+  return formatDateFor(message1) === formatDateFor(message2);
+}
+
+function formatDateFor(message) {
+  var date = new Date(Math.floor(message.ts)*1000);
```
Comment (frapontillo) [inline node-server/server/widget/slacker/slacker.js]: Space between math ops please

Comment (florianmski) [inline node-server/server/widget/slacker/slacker.js]: You mean around the `*` right?

**Comment by florianmski** [comment]
@frapontillo That was part of [PT-325](https://novoda.atlassian.net/browse/PT-325)
This just simplify development setup by avoid typing a few commands, I could have those as a local alias and share it on the #pt though. 
Thoughts? cc @xrigau

File: node-server/server/widget/slacker/slacker.js (hunk 1 above)
Comment (frapontillo) [inline node-server/server/widget/slacker/slacker.js]: Yessir

File: run_da.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+$TERMINAL -e bash -c "cd node-server && npm start"&
+$TERMINAL -e bash -c "cd angular2-client && npm start"&
```
Comment (ouchadam) [inline run_da.sh]: `npm start` or `ng serve` for the angular client?

File: node-server/server/widget/slacker/slacker.js [hunk 3]
```diff
@@ -28,14 +27,22 @@ function messageHandler(self, callback) {
       console.log('message ignored');
       return;
     }
-    if (self.index === 100) {
-      self.index = 0;
+    if (self.messages.length > 0 && !isSameDayFor(message, self.messages[0])) {
+      self.messages.length = 0;
```
Comment (frapontillo) [inline node-server/server/widget/slacker/slacker.js]: YES. XD

Comment (florianmski) [inline node-server/server/widget/slacker/slacker.js]: [Yes, it does](http://stackoverflow.com/a/1232046)
JS :trollface:

**Comment by xrigau** [comment]
>Also add a script to start automatically server and client (not sure how this would work on OSX, sorry).

**Comment by florianmski** [comment]
@xrigau yeah tbh I think I'll remove this script, I just realized that when I need to stop the client or the server it also close the terminal which I need then to reopen, etc... which is really annoying.
I then need to use the `--hold` flag which is completely dependent on the terminal I use.
I think it's best if everyone just use a custom script for their own setup, starting terminal is just too dependent on your config :/

File: node-server/server/widget/slacker/slacker.js (hunk 1 above)
Comment (xrigau) [inline node-server/server/widget/slacker/slacker.js]: datttt formatting

File: run_da.sh (hunk 2 above)
Comment (xrigau) [inline run_da.sh]: script name should be something else. remember this is open source 😄  and would be nice at least 1 line of documentation

also once you start it how do you stop it? perhaps we need some way to tear it down

File: node-server/server/widget/slacker/slacker.js (hunk 3 above)
Comment (takecare) [inline node-server/server/widget/slacker/slacker.js]: woop woop 😄

File: run_da.sh (hunk 2 above)
Comment (takecare) [inline run_da.sh]: what's `da`? 🤔

Comment (florianmski) [inline run_da.sh]: That's a stupid word play 😄 

run da.sh... board

Anyway I'll remove the script as explained in another comment 👍

**Pull request merged**